src/org/labkey/test/components/glassLibrary/navigation/FMBreadCrumbs.java

```diff
@@ -0,0 +1,210 @@
+
+    public FMBreadCrumbs(WebDriver driver)
+    {
+        this(Locators.component.findElement(driver), driver);
+    }
+
+    private FMBreadCrumbs(WebElement element, WebDriver driver)
+    {
+        _breadCrumbsElement = element;
+        _driver = driver;
+    }
+
+    @Override
+    protected WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _breadCrumbsElement;
+    }
+
+    /**
+     * Private function that returns all of the li/span elements in this ol element.
+     *
+     * @return List of span tags in the ordered list.
+     */
+    private List<WebElement> getBreadCrumbElements()
+    {
+        return Locators.breadCrumb.findElements(this);
+    }
+
+    /**
+     * Return the text value of the breadcrumbs.
+     *
+     * @return List containing the text of the breadcrumbs.
+     */
+    public List<String> getBreadCrumbs()
+    {
+        List<String> breadCrumbs = new ArrayList<>();
+        for(WebElement crumb : getBreadCrumbElements())
+        {
+            breadCrumbs.add(crumb.getText());
+        }
+
+        return breadCrumbs;
+    }
+
+    /**
+     * <p>
+     *     A bread crumb may or may not be a link. If it is a link use this to click it.
+     * </p>
+     * <p>
+     *     It is possible that clicking a bread link will cause a navigation event. However this method makes no
+     *     assumptions as to what the click action will be.
+     * </p>
+     *
+     * @param breadCrumbText The text of the breadcrumb.
+     */
+    public void clickBreadCrumbLink(String breadCrumbText)
+    {
+        elementCache().breadCrumbLink(breadCrumbText).click();
+    }
+
+    /**
+     * Check to see if the given breadcrumb has a dropdown option.
+     *
+     * @param breadCrumbText The text of the breadcrumb.
+     * @return Return true if the dropdown button is visible, false otherwise.
+     */
+    public boolean breadCrumbHasDropdown(String breadCrumbText)
+    {
+        // Use a LazyWebElement to find the dropdown. Using a LazyWebElement will return false for isDisplayed if the
+        // element is not there, other finders throw a "No Such Element" exception.
+        return Locators.dropDownButton.findWhenNeeded(elementCache().breadCrumb(breadCrumbText)).isDisplayed();
+    }
+
+    /**
+     * Get the text from the breadcrumb's dropdown menu.
+     *
+     * @param breadCrumbText The text of the breadcrumb.
+     * @return A list of strings with the text from the dropdown menu.
+     */
+    public List<String> getBreadCrumbDropdownText(String breadCrumbText)
+    {
+        WebElement breadCrumb = elementCache().breadCrumb(breadCrumbText);
+        DropdownButtonGroup dropDownMenu = elementCache().dropDownMenu(breadCrumb);
+        if(!dropDownMenu.isExpanded())
+        {
+            dropDownMenu.expand();
+        }
+        return elementCache().dropDownMenu(breadCrumb).getMenuItemTexts();
```

Comment:
It looks like `getMenuItemTexts` will open the menu if needed.
```suggestion
        return elementCache().dropDownMenu(breadCrumb).getMenuItemTexts();
```

Reply:
Removed redundant code.